pypeit/core/slitdesign_matching.py

```diff
@@ -0,0 +1,226 @@
+        if x_det.shape[0] > 10:
+            offs1 = (join[sind[((w1[0] + 1)>(nj - 1)).choose((w1[0] + 1),(nj - 1))]] - x_det_lag)
+            offs2 = (x_det_lag - join[sind[((w1[0] - 1)<0).choose(w1[0] - 1,0)]])
+            offs = (offs1 > offs2).choose(offs1, offs2)
+        else:
+            # [IDL-version comment] so added this brute-force version - still not quite right,
+            # as assumes don't match 2 x_det's to the same x_model
+            offs=numpy.zeros((x_det.shape[0]))
+            for xelement in range(x_det.shape[0]):
+                offs[xelement]=numpy.min(numpy.abs(x_det_lag[xelement]-x_model_trim))
+
+            # use only nbest best matches
+            soffs = numpy.argsort(numpy.abs(offs))
+            nbest2= nbest if nbest<x_det.shape[0] else x_det.shape[0]
+            offs = offs[soffs[0:nbest2]]
+
+        sdev[j] = (offs**2).sum()          # big if match is bad
+
+    best_sdev = int(numpy.mean(numpy.where(sdev==sdev.min())[0]))      # average ind
+    #sdev = sdev[best_sdev]
+
+    return xlag[best_sdev]
+
+
+def discrete_correlate_match(x_det, x_model, step=1, xlag_range=[-50, 50]):
+    """
+    Taken from discrete_correlate_match.pro in DEEP2/spec2d/pro/
+    x_det==x1, x_model==x2_in
+    """
+    best_off = best_offset(x_det, x_model, step=step, xlag_range=xlag_range)
+    x_model_new = x_model - best_off
+
+    ind = numpy.zeros(x_det.shape[0], dtype=int)
+    for i in range(ind.shape[0]):
+        ind[i] = numpy.where(numpy.min(numpy.abs(x_det[i] - x_model_new)) == numpy.abs(x_det[i] - x_model_new))[0]
+
+    dx = x_det - x_model_new[ind]
+    coeff, cov = curve_fit(linefit, x_det, dx, p0=[0, 1])
+    scale = 1 + coeff[1] if x_det.shape[0] > 4 else 1
+    x_model_new *= scale
+
+    new_best_off = best_offset(x_det, x_model_new, step=step, xlag_range=xlag_range)
+
+    x_model_new -= new_best_off
+
+    for i in range(ind.shape[0]):
+        ind[i] = numpy.where(numpy.min(numpy.abs(x_det[i] - x_model_new)) == numpy.abs(x_det[i] - x_model_new))[0]
+
+    dx = x_det - x_model_new[ind]
+    x_model_new += numpy.median(dx)
+
+    for i in range(ind.shape[0]):
+        ind[i] = numpy.where(numpy.min(numpy.abs(x_det[i] - x_model_new)) == numpy.abs(x_det[i] - x_model_new))[0]
+
+    return ind
+
+
+def slit_match(x_det, x_model, step=1, xlag_range=[-50,50], print_matches=False, edge=None):
+    """
+    Taken from DEEP2/spec2d/pro/deimos_slit_match.pro
+    """
+
+    ind=discrete_correlate_match(x_det, x_model, step=step, xlag_range=xlag_range)
+
+    residual = (x_det-x_model[ind]) - numpy.median(x_det-x_model[ind])
+    weights = numpy.zeros(residual.shape[0], dtype=int)
+    weights[numpy.abs(residual) < 100.] = 1
+    if weights.sum()==0: weights=numpy.ones(residual.shape[0], dtype=int)
+    coeff, cov= curve_fit(linefit, x_model[ind]*weights, x_det, p0=[0,1])
+    coeff=numpy.append(coeff, 0)
+    yfit = slit_coeff_eval(x_model[ind], x_model[ind]*0., coeff)
```

Comment:
Can we use np.polyfit here instead of the curve_fit. This is the more pythonic way to fit a straight line. Also, if you need to reject outliers, we should consider using robust_fit.py

Reply:
Made some changes here to use instead `fitting.robust_fit`. I also removed `slit_coeff_eval`

Reply:
I still see the curve fit call above?